Gems/ROS2/Code/Source/RobotImporter/Pages/FileSelectionPage.cpp

```diff
@@ -22,7 +22,6 @@ namespace ROS2
         , m_sdfAssetBuilderSettings(AZStd::make_unique<SdfAssetBuilderSettings>())
     {
         m_fileDialog = new QFileDialog(this);
-        m_fileDialog->setDirectory(QString::fromUtf8(AZ::Utils::GetProjectPath().data()));
```

Comment:
Actually instead of removing setting the directory, I think it would be better to load the most recently used directory into the File Dialog.

We can store the directory being shown into the Editor Preferences section(`/O3DE/Preferences`) of the Settings Registry using the [QFileDialog::directory](https://doc.qt.io/qt-6/qfiledialog.html#directory) function to query it.

We can do this based on the `completeChanged` signal that gets emitted below when a file is selected or a file path is directly typed into the text editor

Reply:
Great suggestion, the only tweak is that I'd be inclined to follow a similar implementation to the AssetImporterManager and store the directory into QSettings instead of the editorpreferences.setreg. Looking at the other settings in the editor preferences, I feel that something like this would just be clutter and doesn't really need to be exposed to the user.

Reply:
Well I don't know where QSettings are stored on Linux for a user. Perhaps it is a location like (`~/.cache/qt/...')?

I agree it is not an editor preferences settings, but more like Editor runtime metadata that is populated based on users actions.

Reply:
The QSettings are being stored in `~/.config/O3DE/O3DE Editor.conf` .

I've changed this to make the file dialog default to the last selected file and directory, but only when the OK button is pressed. In addition, if you change the file in the text box and then open the file dialog, it now defaults to that file correctly too. If the file doesn't exist, it will fall back to defaulting back to the project path again.

Reply:
Sounds good to me!